sdk/inkrecognizer/src/InkRecognitionUnitBuilder.ts

```diff
@@ -0,0 +1,368 @@
+    unit.alternates = [];
+    json.alternates.forEach((jsonLineAlternate: JsonLineAlternate) => {
+      unit.alternates.push(jsonLineAlternate.recognizedString);
+    });
+  }
+
+  unit.toJSON = () => {
+    let jsonUnit = getInkRecognitionUnitJSON(unit);
+    let jsonThis = {
+      text: unit.text,
+      alternates: unit.alternates,
+    };
+    return { ...jsonUnit, ...jsonThis };
+  };
+  return unit;
+}
+
+function createParagraph(json: JsonInkRecognitionUnit): Paragraph {
+  let unit = createInkRecognitionUnit(json) as Paragraph;
+  return unit;
+}
+
+function createWritingRegion(json: JsonInkRecognitionUnit): WritingRegion {
+  let unit = createInkRecognitionUnit(json) as WritingRegion;
+  return unit;
+}
+
+function processInkRecognitionUnit(json: JsonInkRecognitionUnit, result: InkRecognitionResult): InkRecognitionUnit {
+  let unit: InkRecognitionUnit = {} as InkRecognitionUnit;
+  switch (json.category) {
+    case 'inkDrawing':
+      unit = createInkDrawing(json as JsonInkDrawing);
+      result.inkDrawings.push(unit as InkDrawing);
+      break;
+    case 'inkWord':
+      unit = createInkWord(json as JsonInkWord);
+      result.inkWords.push(unit as InkWord);
+      break;
+    case 'line':
+      unit = createLine(json as JsonLine);
+      result.lines.push(unit as Line);
+      break;
+    case 'paragraph':
+      unit = createParagraph(json);
+      result.paragraphs.push(unit as Paragraph);
+      break;
+    case 'writingRegion':
+      unit = createWritingRegion(json);
+      result.writingRegions.push(unit as WritingRegion);
+      break;
+    case 'inkBullet':
+      unit = createInkBullet(json as JsonInkBullet);
+      result.inkBullets.push(unit as InkBullet);
+      break;
+    default:
+      unit = createInkRecognitionUnit(json);
+      break;
+  }
+  return unit;
+}
+
+export function parseInkRecognitionResult(responseText: string): InkRecognitionResult {
+  let idToRecognitionUnitMap: Map<Number, InkRecognitionUnit> = new Map<Number, InkRecognitionUnit>();
+  let idToJsonRecognitionUnitMap: Map<Number, JsonInkRecognitionUnit> = new Map<Number, JsonInkRecognitionUnit>();
+
+  let result = {} as InkRecognitionResult;
+  result.inkDrawings = [];
+  result.inkWords = [];
+  result.inkBullets = [];
+  result.lines = [];
+  result.paragraphs = [];
+  result.writingRegions = [];
+
+  // Parse InkRecognitionUnits from json
+  const json: JsonInkRecognitionResult = JSON.parse(responseText) as JsonInkRecognitionResult;
+  json.recognitionUnits.forEach((jsonUnit: JsonInkRecognitionUnit) => {
+    let unit = processInkRecognitionUnit(jsonUnit, result);
+    idToRecognitionUnitMap.set(unit.id, unit);
+    idToJsonRecognitionUnitMap.set(unit.id, jsonUnit);
+  });
+
+  // Fill parent and children recognitionUnits
+  idToRecognitionUnitMap.forEach((value: InkRecognitionUnit, key: Number) => {
+    if (idToJsonRecognitionUnitMap.has(key)) {
+      let jsonUnit = idToJsonRecognitionUnitMap.get(key);
+      if (jsonUnit) {
```

Comment:
What do you think of doing an early return here? It might help with the readability of the code.

Reply:
Sure, sounds good idea.